cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/StructuredRecord.java

```diff
@@ -185,6 +186,10 @@ public BigDecimal getDecimal(String fieldName) {
     }
 
     int scale = logicalTypeSchema.getScale();
+    if (value instanceof ByteBuffer) {
+      return new BigDecimal(new BigInteger(((ByteBuffer) value).array()), scale);
```

Comment:
array() won't always give the correct byte array. Should use the Bytes.toBytes() method.

Reply:
Use `io.cdap.cdap.api.common.Bytes.toBytes(ByteBuffer)`.

Reply:
Changed to use `io.cdap.cdap.api.common.Bytes.toBytes(ByteBuffer)`.